examples/nlp/language_modeling/megatron_gpt_eval.py

```diff
@@ -182,9 +182,16 @@ def request_data(data):
 
 
 class RequestDataSet(Dataset):
-    def __init__(self, sentences):
+    def __init__(self, sentences, prompt_learning_inference=False):
         super().__init__()
-        self.sentences = sentences
+        if prompt_learning_inference:
+            self.sentences = []
+
+            for dataset in sentences:
+                for line in open(dataset, 'r', encoding='utf-8').readlines():
+                    self.sentences.append(line)
```

Comment:
Here you still load all the sentences from the files into the memory. Why it avoids OOM?
Also consider add the doc string for RequestDataSet. the sentences can be multiple things now.

Reply:
Ended up making a separate script for prompt learning inference and removing the use of request dataset for prompt learning inference all together. 

Previously, the inference code was trying to load and run inference on the entire test set at once without breaking it into batches. This was causing OOM for large test sets. This change was to make sure the data was loaded and passed to the model in batches.